clib/cList.mli

```diff
@@ -186,6 +186,10 @@ val fold_left3 : ('a -> 'b -> 'c -> 'd -> 'a) -> 'a -> 'b list -> 'c list -> 'd
 (** Like [List.fold_left] but for 3 lists; raise [Invalid_argument _] if
     not all lists of the same size *)
 
+val fold_left4 : ('a -> 'b -> 'c -> 'd -> 'e -> 'a) -> 'a -> 'b list -> 'c list -> 'd list -> 'e list -> 'a
```

Comment:
Honestly I don't think it's a great idea to have these combinators. They're very ad-hoc and end up making the code harder to understand when there are that many arguments. I'd rather have an explicit fixpoint where these are used. I know we've been introducing those, but in my opinion beyond two arguments it starts to be counter-productive.

Reply:
The motivation was to comply with the idiomatic style used in `comFixpoint.interp_recursive` (in today's [master](https://github.com/coq/coq/blob/fc1b697e47f9593b730f105be88e9260edbcb851/vernac/comFixpoint.ml#L178-L234)) (in the current version of the ongoing work on merging the different execution paths for `Fixpoint` in [#18811](https://github.com/coq/coq/blob/43ab803122d2d774bd378fe634112345b583b335/vernac/comFixpoint.ml#L408-L459)). In that code, the arguments have meaningful names, so it is roughly clear which one is which one. An alternative would be to use lists of tuples rather than tuples of lists. I think about it but the tuples involved are not always the same selection among the various attributes `name`, `relevance`, `binders`, `body`, `type`, `impls`, `wf_order`, `notations`.

PS: Since the code in #18811 will eventually lead to a relatively "completely new" implementation, I'd be happy to have it done in a way which everyone found satisfying enough for the next coming years.

Reply:
So do we merge or not?